src/Data.cpp

```diff
@@ -1240,27 +1357,18 @@ bool Data::LoggedIn() const
   return m_manager.loggedIn();
 }
 
-properties_t Data::StreamProperties(const std::string & url, const std::string & streamType, bool isLive)
+void Data::StreamProperties(std::vector<kodi::addon::PVRStreamProperty>& properties, const std::string & url, const std::string & streamType, bool isLive)
```

Comment:
Why did you move the return type to be parameter?

Reply:
Was to remove some not needed malloc's and free's in C++ background, as by return it recreate all data to store on his parent call and delete the ones from return.

Reply:
> as by return it recreate all data to store on his parent call and delete the ones from return.

This is definitely not the case up from C++11 (move constructors etc..)